### PR TITLE
Miscellaneous devicetree improvements and cleanups

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
This is a subset of the patches in https://github.com/zephyrproject-rtos/zephyr/pull/52272 that are standalone improvements and can IMO be merged now.

### CODE REVIEW

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by galak — CHANGES_REQUESTED** [review]
Can you pull the cmake change into its own PR.  Thus @tejlmand can review that separately from the python lib changes.

**Review by galak — CHANGES_REQUESTED** [review]
Can you also split out the 'python-devicetree: re-work reg and ranges handling ' commit into its own PR.  All the other cleanup commits look good and will +1 (after the minor fix that was already mentioned).

Adding 'reg' and 'ranges' to dtlib.py concerns me as dtlib doesn't generally have much understanding of the contents of the tree.  I think it best to try and maintain that abstraction/separation.  I'd rather see 'reg' and 'ranges' add to edtlib.py.

**Comment by mbolivar-nordic** [comment]
> I'd rather see 'reg' and 'ranges' add to edtlib.py.

That would defeat the purpose of being able to use them in sysdtlib, though, where we aren't using edtlib (because we don't need any bindings), but we still need to understand the DTSpec portions of the tree (to rewrite registers).

File: scripts/dts/python-devicetree/src/devicetree/dtlib.py [hunk 1]
```diff
@@ -310,7 +310,7 @@ def __init__(self, node: Node, name: str):
     @property
     def type(self) -> Type:
         """
-        See the class docstring.
+        See the class documentation.
```
[the remark this answers is not in the document]
Comment (mbolivar-nordic) [reply scripts/dts/python-devicetree/src/devicetree/dtlib.py:458]: @nordicjm I don't see a problem. This is both documentation and a string, so both are correct. "Docstring" is a python term for this (see https://peps.python.org/pep-0257/), but the point of this commit is to make this docstring match all the other docstrings in this module. These all say "See the class documentation." and this commit is making this docstring match the rest of them.

**Review by galak — DISMISSED** [review]

**Comment by mbolivar-nordic** [comment]
@galak sorry, I caught an issue and pushed another change

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
> @galak sorry, I caught an issue and pushed another change

Hmm, what's up with the scancode license?

**Comment by mbolivar-nordic** [comment]
> Hmm, what's up with the scancode license?

I asked about this in `#ci` yesterday but got no response

**Comment by stephanosio** [comment]
> > @galak sorry, I caught an issue and pushed another change
> 
> Hmm, what's up with the scancode license?

Nothing. License check failure on any non-Apache 2.0 files being added to the tree is normal. If sanctioned, the check failure can be overridden.

**Comment by mbolivar-nordic** [comment]
> License check failure on any non-Apache 2.0 files being added to the
> tree is normal. If sanctioned, the check failure can be overridden.

This code is being copy/pasted from a file which already has a BSD 3
clause license, so we have to keep this license.

**Review by nordicjm — APPROVED** [review]

**Pull request merged**